api/src/main/java/org/eclipse/microprofile/reactive/streams/ReactiveStreams.java

```diff
@@ -231,4 +232,23 @@ private ReactiveStreams() {
                                                  PublisherBuilder<? extends T> b) {
         return new PublisherBuilder<>(new Stage.Concat(a.toGraph(), b.toGraph()));
     }
+
+    /**
+     * Creates a publisher from a {@link CompletionStage}.
+     * <p>
+     * <img src="doc-files/fromCompletionStage.png" alt="fromCompletionStage marble diagram">
+     * <p>
+     * When the {@code CompletionStage} is redeemed, the publisher will emit the redeemed element, and then signal
+     * completion. If the completion stage is redeemed with {@code null}, no element will be emitted, the publisher
+     * will immediately signal completion.
+     * <p>
+     * If the {@code CompletionStage} is completed with a failure, this failure will be propagated through the stream.
+     *
+     * @param completionStage The {@code CompletionStage} to create the publisher from.
+     * @param <T> The type of the {@code CompletionStage} value.
+     * @return A {@code PublisherBuilder} representation of this {@code CompletionStage}.
+     */
+    public static <T> PublisherBuilder<T> fromCompletionStage(CompletionStage<? extends T> completionStage) {
```

Comment:
You could have a `Supplier<Optional<? extends T>> whenNull`—and if that returns null, you complete with NPE, if that is empty you only complete, and if that has a value, you use that instead (unless that value is null, in which case it is an NPE).